pkg/compactor/compactor.go

```diff
@@ -384,6 +389,10 @@ func newMultitenantCompactor(
 			Name: "cortex_compactor_compaction_interval_seconds",
 			Help: "The configured interval on which compaction is run in seconds. Useful when compared to the last successful run metric to accurately detect multiple failed compaction runs.",
 		}),
+		outOfSpace: promauto.With(registerer).NewGauge(prometheus.GaugeOpts{
+			Name: "cortex_compactor_out_of_space_errors_total",
+			Help: "Number of times a compaction failed because the compactor was out of space.",
```

Comment:
[nit] Have you considered adding "disk" as well?

```suggestion
			Name: "cortex_compactor_disk_out_of_space_errors_total",
			Help: "Number of times a compaction failed because the compactor disk was out of space.",
```